department-of-veterans-affairs/abd-vro
# EP Merge App: persistent job store

## What was the problem?
The current merge app stores information about merge jobs in-memory, which could lead to jobs being lost if the app is suddenly terminated.

Associated tickets or Slack threads:
- #2197 

## How does this fix it?[^1]
[description of how things will work after this PR](https://github.com/department-of-veterans-affairs/abd-vro/issues/2196#issuecomment-1828608254)

## How to test this PR
- Apply the database migration by running the db-init container. You may have to rebuild the docker image to pick up the new .sql file, e.g. `docker rm db-init && ./gradlew :db-init:dockerStart`
- in `domain-ee/ee-ep-merge-app`: run `pytest` and `pytest ./integration`
- **NOTE**: the tests in this PR are relatively low effort and need work. In particular, they don't cover most of the logic of `JobStore.init` nor do they cover cases where jobs are restarted from the app's `on_start_up` method. Feedback on how best to mock/simulate these cases would be greatly appreciated!


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
